src/main/scala/temple/generate/server/go/service/GoServiceHookGenerator.scala

```diff
@@ -6,35 +6,53 @@ import temple.generate.CRUD.{CRUD, presentParticiple}
 import temple.generate.server.AttributesRoot.ServiceRoot
 import temple.generate.server.go.common.GoCommonGenerator._
 import temple.generate.server.go.common.GoCommonHookGenerator
-import temple.generate.utils.CodeTerm.mkCode
+import temple.generate.utils.CodeTerm.{CodeWrap, mkCode}
+import temple.utils.StringUtils.doubleQuote
 
 object GoServiceHookGenerator {
 
-  private def generateBeforeHookType(root: ServiceRoot, operation: CRUD): String = operation match {
-    case CRUD.List =>
-      root.readable match {
-        case Readable.This =>
-          s"func(env *env, input *dao.List${root.name}Input) *HookError"
-        case Readable.All =>
-          s"func(env *env) *HookError"
-      }
-    case op @ (CRUD.Read | CRUD.Delete | CRUD.Identify) =>
-      s"func(env *env, input *dao.${op.toString}${root.name}Input) *HookError"
-    case op @ (CRUD.Create | CRUD.Update) =>
-      if (root.requestAttributes.isEmpty)
-        s"func(env *env, input *dao.${op.toString}${root.name}Input) *HookError"
-      else
-        s"func(env *env, req ${op.toString.toLowerCase}${root.name}Request, input *dao.${op.toString}${root.name}Input) *HookError"
+  private[service] def generateImports(root: ServiceRoot): String = {
+    val daoImport = s"${doubleQuote(root.module + "/dao")}"
+    if (root.projectUsesAuth)
+      mkCode("import", CodeWrap.parens.tabbed(daoImport, s"${doubleQuote(root.module + "/util")}"))
+    else mkCode("import", daoImport)
+  }
+
+  private def generateBeforeHookType(root: ServiceRoot, operation: CRUD): String = {
+    val authArg =
+      if (root.projectUsesAuth) ", auth *util.Auth"
```

Comment:
Hmm, how about rewriting this using `.prefix` and `.parens` from CodeWrap?

Reply:
I knew you were gonna make this comment, the reason I didn't is because I couldn't be arsed- but I'll do it 😉